compiler-rt/lib/radsan/tests/radsan_test_utilities.h

```diff
@@ -0,0 +1,47 @@
+/**
+    This file is part of the RealtimeSanitizer (RADSan) project.
+    https://github.com/realtime-sanitizer/radsan
+
+    Copyright 2023 David Trevelyan & Alistair Barker
+    Subject to GNU General Public License (GPL) v3.0
+*/
+
+#pragma once
+
+#include "gmock/gmock.h"
+#include "radsan.h"
+#include <string>
+
+namespace radsan_testing {
+
+template <typename Function>
+void realtimeInvoke(Function &&func) 
+{
+  radsan_realtime_enter();
```

Comment:
Note I adjusted this to call the things directly (as we don't have the codegen portion in this reivew)